### DISCUSSION TITLE
creating-an-extension-method-for-attaching-key-value-pairs-to-scope-state-using-asp-net-core/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#18] [Announcements]
# creating-an-extension-method-for-attaching-key-value-pairs-to-scope-state-using-asp-net-core/

In this post I describe a small extension method that makes logging scopes as key-value pairs in ASP.NET Core with Serilog and Seq a little nicer.

https://andrewlock.net/creating-an-extension-method-for-attaching-key-value-pairs-to-scope-state-using-asp-net-core/

**Comment by lastsignal** [comment]
Great like always. 

Alternatively, we can do this:

``` csharp

public statis IDisposable BeginScope(this ILogger logger, params (string key, object value)[] paris)
{
    return logger.BeginScope(pairs.ToDictionary(p=>p.key, p=>p.value));
}
```

usage:

``` csharp
using (_logger.BeginScope(
    (nameof(productId), productId),
    (nameof(basketId), basketId)))
{
    ...
}

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at September 20 2017, 08:51, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/18#discussioncomment-6042942)</em>

---

<p>Thanks for the comment as always :)</p><p>That's something I do too, but I was looking at ways of simplifying the usage of the standard ASP.NET Core Logging abstractions, in particular as part of the main app code, rather than at log configuration time. These can then be used in standard libraries, without depending on Serilog to add arbitrary extra values, but sticking to the Serilog/Seq semantics (which make a lot of sense)</p>

[a comment by an automated account is not shown]
